--- a/tools/initialize_database.py	2012-09-10 22:59:43.625005550 +1100
+++ b/tools/initialize_database.py	2012-09-10 23:00:44.285006986 +1100
@@ -50,7 +50,7 @@
     from karesansui.lib.utils import generate_phrase, generate_uuid, string_from_uuid, is_uuid
     from karesansui.lib.const import DEFAULT_LANGS
 
-    reg_email = re.compile("^[a-zA-Z0-9\./-]{1,}@[a-zA-Z0-9\./-]{4,}$")
+    reg_email = re.compile("^[a-zA-Z0-9\./_-]{1,}@[a-zA-Z0-9\./-]{4,}$")
     if opts.email:
         if reg_email.search(opts.email) is None:
             raise Exception('ERROR: Illigal option value. option=%s value=%s' % ('-m or --email', opts.email))
