--- collectd-4.10.0/src/collectd.conf.in	2010-05-01 11:15:45.000000000 +0200
+++ collectd-4.10.0/src/collectd.conf.in	2010-09-19 15:00:15.893887909 +0200
@@ -116,7 +116,7 @@
 #@BUILD_PLUGIN_PYTHON_TRUE@LoadPlugin python
 #@BUILD_PLUGIN_ROUTEROS_TRUE@LoadPlugin routeros
 #@BUILD_PLUGIN_RRDCACHED_TRUE@LoadPlugin rrdcached
-@LOAD_PLUGIN_RRDTOOL@LoadPlugin rrdtool
+#@LOAD_PLUGIN_RRDTOOL@LoadPlugin rrdtool
 #@BUILD_PLUGIN_SENSORS_TRUE@LoadPlugin sensors
 #@BUILD_PLUGIN_SERIAL_TRUE@LoadPlugin serial
 #@BUILD_PLUGIN_SNMP_TRUE@LoadPlugin snmp
@@ -758,6 +758,7 @@
 #	CreateFiles true
 #	CollectStatistics true
 #</Plugin>
+Include "/etc/collectd.d"
 
 #<Plugin rrdtool>
 #	DataDir "@prefix@/var/lib/@PACKAGE_NAME@/rrd"
